File: spring-webflux/src/main/java/software/darkmatter/school/blog/api/dto/PostDto.java
package software.darkmatter.school.blog.api.dto;

import java.time.OffsetDateTime;

public record PostDto(
    Long id,
    String title,
    String summary,
    String text,
    OffsetDateTime createdAt,
    UserDto createdBy,
    OffsetDateTime updatedAt,
    OffsetDateTime publishedAt
) {}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/api/dto/PostDto.kt
package software.darkmatter.school.blog.api.dto

import java.time.OffsetDateTime

data class PostDto(
    val id: Long,
    val title: String?,
    val summary: String?,
    val content: String?,
    val createdAt: OffsetDateTime,
    val createdBy: UserDto,
    val updatedAt: OffsetDateTime,
    val publishedAt: OffsetDateTime?
)
